test/nftTransfer.js

```diff
@@ -112,7 +112,7 @@ describe("Token Transfer", function () {
         }
       } else {
         const txPromise = nftFeature.from(owner1)
-          .transferNFT(wallet1.contractAddress, nftContract.contractAddress, recipientAddress, nftId, safe, ZERO_BYTES32);
+          .transferNFT(wallet1.contractAddress, nftContract.contractAddress, recipientAddress, nftId, safe, ZERO_BYTES32, { gasLimit: 300000 });
```

Comment:
Have you benchmarked the extra gas cost for static calls that are initiated on-chain (e.g. safe NFT transfers)?

Reply:
It adds 110k gas to static calls

Reply:
110k??? Is it for the entire static call, or on top of what we had in the earlier implementation (e.g. 1.6)?

Reply:
I managed to reduce it to 1-2k by limiting the gas sent to the inner call. It looks like a write operation in a static context  cause an `assert` instead of a `revert`.

Reply:
Given that it only saves 1-2k of gas I think we shouldn't specify the gas passed to the call because it will start failing for non-static calls the day the cost of the CALL opcode is increased.

Reply:
Oh no it does save saves 99% of the gas. I meant that the *increase* (relative to 1.6) of the gas cost is now only 1-2k instead of 110k :)

Reply:
Ok that's a lot then :-)

But shouldn't the minimum gas stipend for a call be 2300? I just want to make sure we provide enough gas for the call to fail because of the non-static context, and not for any other reason (specifically an out of gas).

Reply:
The logic of the subcall costs 425 gas to execute. In my tests I found that passing 625 gas to the subcall was enough to make it succeed. I'm not sure why 200 gas extra is required but it seems like the cost of the CALL itself does not need to be included in the stipend. A stipend of 1000 leaves a margin of 1000-625=375 gas to account for opcode repricing. I could increase it but keep in mind that all the gas passed to the subcall is fully consumed by the `assert`. 

Also note that if the stipend was too little for the subcall to succeed in a non-static context, you would notice the last test of versionManager.js fail

Reply:
Anyways, I've increased the stipend to 3000 to be more comfy.